File: shared/waveRoom.ts
/** Represents the current state of a wave room */
export enum WaveRoomState {
    WAITING,
    STARTED,
    ENDED,
}

/** Determines if a wave room is public or private */
export enum WaveRoomVisibleState {
    PUBLIC,
    PRIVATE,
}

export const VISIBLE_STATE_VALUES = Object.values(WaveRoomVisibleState);

/** Indicates if a player is ready to start the wave */
export enum WaveRoomPlayerReadyState {
    UNREADY,
    READY,
}

export const PLAYER_STATE_VALUES = Object.values(WaveRoomPlayerReadyState);

File: Sources/Client/private/src/Florr/PseudoNative/Wave/WaveRoom.zig
/// Represents the state of the wave room.
pub const RoomState = enum(u8) {
    /// Waiting, meaning waiting for players (not starting).
    waiting,
    /// Playing, meaning playing currently.
    playing,
    /// Ended, meaning game is already over.
    ended,
};

/// Represents the visible state of the wave room.
pub const RoomVisibleState = enum(u8) {
    public,
    private,
};

/// Represents the player ready state in the wave room.
pub const RoomPlayerReadyState = enum(u8) {
    preparing,
    ready,
};
